internal/process/provisioning/create_runtime_resource_step.go

```diff
@@ -250,6 +251,49 @@ func (s *CreateRuntimeResourceStep) createShootProvider(operation *internal.Oper
 			VolumeSize: fmt.Sprintf("%sGi", volumeSize),
 		}
 	}
+
+	if len(operation.ProvisioningParameters.Parameters.AdditionalWorkerNodePools) > 0 {
+		additionalWorkerNodePoolsMaxUnavailable := intstr.FromInt32(int32(0))
+		workers := make([]gardener.Worker, 0, len(operation.ProvisioningParameters.Parameters.AdditionalWorkerNodePools))
+
+		for _, additionalWorkerNodePool := range operation.ProvisioningParameters.Parameters.AdditionalWorkerNodePools {
+			workerZones := values.Zones
+			if !additionalWorkerNodePool.HAZones {
```

Comment:
So if there is no HA, we pick up one zone randomly from previously randomly assigned zones for the cluster.
Am I correct?

So far we used random.shuffle. Please consider this approach as "conventional":
```
	rand.Shuffle(len(zones), func(i, j int) { zones[i], zones[j] = zones[j], zones[i] })
	return zones[:zonesCount]
```